rust/src/common.rs

```diff
@@ -72,3 +72,12 @@ pub unsafe extern "C" fn rs_cstring_free(s: *mut c_char) {
     }
     drop(CString::from_raw(s));
 }
+
+/// Convert an u8-array of data into a hexadecimal representation
+pub fn to_hex(input: &[u8]) -> String {
```

Comment:
@jasonish any thoughts on this name?

Reply:
Maybe it should be something like `slice_to_hex`

Reply:
How about `byteslice_to_hex`?

Reply:
I think the name is OK if its the one true format of hex we want to support.  As most things we'd want to print as hex will already by a Vec<u8> or a &[u8], or have some sort of as_bytes() method (strings).

How does this differ from doing the builtin `let hex_string = format!("{:x}", some_byte_slice);`?

Reply:
Your example does not build for me:
```rust
let byteslice: &[u8] = tx.hdr.ikev1_header.key_exchange.as_ref();
let hex_string = format!("{:x}", byteslice);
SCLogInfo!("{:?}", hex_string);
```

```
    |
130 |                     let hex_string = format!("{:x}", byteslice);
    |                                                      ^^^^^^^^^ the trait `std::fmt::LowerHex` is not implemented for `[u8]`
    |
    = note: required because of the requirements on the impl of `std::fmt::LowerHex` for `&[u8]`
    = note: required by `std::fmt::LowerHex::fmt`

```

but I just noticed that there is also `to_hex_string(bytes: &[u8]) -> String` in `rust/src/krb/krb5.rs` and that differs only by uppercase vs. lowercase output. Maybe it makes sense to consolidate these in one place...

Reply:
Ah, your right.  The hashing module does this, but the hasing functions return a GenericArray which implement `LowerHex`...  So I'm happy with this as as.

As for `to_hex_string`in krb5, it doesn't appear to be used anywhere.

Reply:
Okay, thanks for your insight! :+1: